### PR TITLE
Fix the tag name for driver builds

### PR DESCRIPTION
**Pull request by saikat-royc**
When a commit is detected with a tag, the tag name should be used
instead of the commit sha.

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
/kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: build/generate_image_tags.sh [hunk 1]
```diff
@@ -35,7 +35,7 @@ if [ $LATEST_GIT_TAG ]; then
 fi
 
 if [ $LATEST_TAG_REV ] && [ $LATEST_TAG_REV == $HEAD_REV ]; then
```
Comment (mattcary) [inline build/generate_image_tags.sh:37]: Do these LATEST_TAG_REV also need to change to the git tag?

Comment (saikat-royc) [reply build/generate_image_tags.sh:37]: no that should be correct since it compares revision(or sha)

Comment (mattcary) [reply build/generate_image_tags.sh:37]: Ah I see you compare with the hashes but need to mark with the tag. Thx

/lgtm

**Pull request merged**